core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/LdapAuthorizationRule.scala

```diff
@@ -74,7 +74,28 @@ object LdapAuthorizationRule {
     override val name = Rule.Name("ldap_authorization")
   }
 
-  final case class Settings(ldap: LdapAuthorizationService,
-                            permittedGroups: UniqueNonEmptyList[Group],
-                            allLdapGroups: UniqueNonEmptyList[Group])
+  final case class Settings(ldap: LdapAuthorizationService, permittedGroups: GroupsLogic, allLdapGroups: UniqueNonEmptyList[Group])
+
+  sealed trait GroupsLogic {
+    val groups: UniqueNonEmptyList[Group]
+  }
+
+  object GroupsLogic {
+    final case class Or(override val groups: UniqueNonEmptyList[Group]) extends GroupsLogic
+
+    final case class And(override val groups: UniqueNonEmptyList[Group]) extends GroupsLogic
+  }
+
+  implicit class GroupsLogicExecutor(val groupsLogic: GroupsLogic) extends AnyVal {
+    def availableGroupsFrom(userGroups: UniqueNonEmptyList[Group]): Option[UniqueNonEmptyList[Group]] = {
+      groupsLogic match {
+        case And(groups) =>
+          val intersection = userGroups intersect groups
+          if (intersection == groups.toSet) Some(groups) else None
```

Comment:
the codacy warn is right - you compare two objects with different types. It's rather dangerous. You can `import cats.implicits._` and use `===` instead of `==` (compiler will fail here)

Reply:
I tried and failed. The === won't pass the tests. Plus, `Intersection` and `groups.toSet` are the same type `Set[Group]`, so == should be ok, right?  🤔